src/test/client/TestClient.h

```diff
@@ -114,6 +121,128 @@ class ClientScaffold : public Client {
     int walk(std::string_view path, struct walk_dentry_result* result, const UserPerm& perms, bool followsym=true) {
       return Client::walk(path, result, perms, followsym);
     }
+
+    void ll_write_n_bytes(struct Fh *fh, size_t to_write, size_t block_size,
+                            int iov_cnt, off_t *offset) {
+      /// @brief Write N bytes of data asynchronously.
+      /// @param fh - File handle
+      /// @param to_write - bytes to write
+      /// @param block_size - Total size of each iovec structs array
+      /// @param iov_cnt - Number of elements in iovec structs array
+      /// @param offset - location to start writing from
+
+      const size_t DATA_PER_BLOCK = size_t(block_size / iov_cnt);
+      // to persist memory address of out_buf after every loop iteration
+      std::vector<std::string> buffers;
+      buffers.reserve(iov_cnt);
+      struct iovec iov_buf_out[iov_cnt];
+
+      for (int i = 0; i < iov_cnt; ++i) {
+        std::string out_buf(DATA_PER_BLOCK, 'a' + i);
+        iov_buf_out[i].iov_base = out_buf.data();
+        iov_buf_out[i].iov_len = DATA_PER_BLOCK;
+        buffers.push_back(std::move(out_buf));
+      }
+
+      C_SaferCond *writefinish = nullptr;
+      int64_t rc = 0, bytes_written = 0;
+
+      while (to_write > 0) {
+        if (to_write >= block_size) {
+          writefinish = new C_SaferCond("writefinish-upto-blocksize");
+          rc = ll_preadv_pwritev(fh, iov_buf_out, iov_cnt, *offset,
+                                true, writefinish, nullptr);
+          ASSERT_EQ(rc, 0);
+          bytes_written = writefinish->wait();
+          ASSERT_EQ(bytes_written, block_size);
+          *offset += bytes_written;
+          to_write -= bytes_written;
+        } else {
+          // Allocate a smaller buffer for the remaining bytes
+          std::string small_buf(to_write, 'z');
+          struct iovec iov_small_buf_out[1] = {
+              {small_buf.data(), to_write}
+          };
+          writefinish = new C_SaferCond("writefinish-small-buffer");
```

Comment:
Is this ever deallocated?

Reply:
https://github.com/ceph/ceph/compare/4135540942dd948cb59520072b0152b7e5afcc72..1fa41e1070f91fbb9d50c6fa42831a7783f69fca